### PR TITLE
CM-664: Zip Files in C#/.NET

### PR DESCRIPTION
**Pull request by MPrtenjak**
New article about zip files

### CODE REVIEW

File: csharp-basic-topics/Zip-files-in-NET/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[*.{cs,vb}]
```
Comment (PhilBroderick) [inline csharp-basic-topics/Zip-files-in-NET/.editorconfig:1]: We shouldn't commit this file. Please remove.

File: csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/Zip-files-in-NET.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <RootNamespace>Zip_files_in_NET</RootNamespace>
```
Comment (PhilBroderick) [inline csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/Zip-files-in-NET.csproj:6]: Solutions/Projects should follow the PascalCase convention.

File: csharp-basic-topics/Zip-files-in-NET/Tests/CreateZipFileTest.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            using var src = ZipFile.OpenRead(sourceZipFile);
+            using var dest = ZipFile.OpenRead(destinationZipFile);
+            var srcFiles = src.Entries.Where(e => !string.IsNullOrEmpty(e.Name)).Select(e => Path.GetFullPath(e.FullName)).ToArray();
+            var destFiles = dest.Entries.Where(e => !string.IsNullOrEmpty(e.Name)).Select(e => Path.GetFullPath(e.FullName)).ToArray();
+            CollectionAssert.AreEquivalent(srcFiles, destFiles);
+        }
+
+        private string sourceZipFile = string.Empty;
+        private string destinationZipFile = string.Empty;
+        private string tempFolder = string.Empty;
```
Comment (PhilBroderick) [inline csharp-basic-topics/Zip-files-in-NET/Tests/CreateZipFileTest.cs:43]: Being picky - but we should define these at the start of the class. Also they should be prefixed with `_`.

File: csharp-basic-topics/Zip-files-in-NET/Tests/ReadZipFilesTest.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+namespace Tests
+{
+    [TestClass]
+    public class ReadZipFilesTest
+    {
+        [TestMethod]
+        [DataRow("multi-folder.zip", 10)]
+        [DataRow("multi-file.zip", 2)]
+        [DataRow("single-file.zip", 1)]
+        public void GivenZipFile_WhenListContent_ThenExpectenNumberOfLinesShouldExists(
```
Comment (PhilBroderick) [inline csharp-basic-topics/Zip-files-in-NET/Tests/ReadZipFilesTest.cs:12]: ThenExpectenNumberOfLinesShouldExists -> ThenExpecte**d**NumberOfLinesShouldExists

File: csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/CreateZipFiles.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            var helloText = "Hello world!";
+
+            // opening a new ZIP file
+            using var archive = ZipFile.Open(zipFileName, ZipArchiveMode.Create);
+
+            // creating an empty entry with a defined name
+            var entry = archive.CreateEntry("hello.txt");
+
+            // using stream to add text to newly created entry
+            using (Stream st = entry.Open())
```
Comment (PhilBroderick) [inline csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/CreateZipFiles.cs:45]: `using var` instead of `using Stream`

**Review by PhilBroderick — CHANGES_REQUESTED** [review]
A few changes required.

File: csharp-basic-topics/Zip-files-in-NET/.editorconfig (hunk 1 above)
Comment (MPrtenjak) [reply csharp-basic-topics/Zip-files-in-NET/.editorconfig:1]: This file is here, so the code is correctly formatted to 4 spaces. That was the requirement from Marinko in the last article, and this file is also in the git of the previous article

File: csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/Zip-files-in-NET.csproj (hunk 2 above)
Comment (MPrtenjak) [reply csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/Zip-files-in-NET.csproj:6]: Corrected

File: csharp-basic-topics/Zip-files-in-NET/Tests/CreateZipFileTest.cs (hunk 3 above)
Comment (MPrtenjak) [reply csharp-basic-topics/Zip-files-in-NET/Tests/CreateZipFileTest.cs:43]: Corrected

File: csharp-basic-topics/Zip-files-in-NET/Tests/ReadZipFilesTest.cs (hunk 4 above)
Comment (MPrtenjak) [reply csharp-basic-topics/Zip-files-in-NET/Tests/ReadZipFilesTest.cs:12]: Corrected

File: csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/CreateZipFiles.cs (hunk 5 above)
Comment (MPrtenjak) [reply csharp-basic-topics/Zip-files-in-NET/Zip-files-in-NET/CreateZipFiles.cs]: Corrected

File: csharp-basic-topics/ZipFilesInNet/ZipFilesInNET/ZipFilesInNET.csproj [hunk 6]
```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <RootNamespace>Zip_files_in_NET</RootNamespace>
```
Comment (PhilBroderick) [inline csharp-basic-topics/ZipFilesInNet/ZipFilesInNET/ZipFilesInNET.csproj:6]: We've still got a reference to the old naming - this should be removed and the namespace/usings updated in any necessary files.

**Review by PhilBroderick — APPROVED** [review]

**Review by PhilBroderick — APPROVED** [review]

**Pull request merged**